include/itkCLEDataManager.hxx

```diff
@@ -0,0 +1,121 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#ifndef itkCLEDataManager_hxx
+#define itkCLEDataManager_hxx
+
+#include "itkImageRegionIterator.h"
+#include "itkImageRegionConstIterator.h"
+
+//#define VERBOSE
+
+namespace itk
+{
+
+template <typename TPixel, unsigned int VImageDimension>
+CLEDataManager<TPixel, VImageDimension>::CLEDataManager()
+{
+  this->Initialize();
+}
+
+template <typename TPixel, unsigned int VImageDimension>
+CLEDataManager<TPixel, VImageDimension>::~CLEDataManager()
+{
+
+}
+
+template <typename TPixel, unsigned int VImageDimension>
+void
+CLEDataManager<TPixel, VImageDimension>::Allocate()
+{
+
+}
+
+template <typename TPixel, unsigned int VImageDimension>
+void
+CLEDataManager<TPixel, VImageDimension>::Initialize()
+{
+  m_IsGPUBufferDirty = false;
+  m_IsCPUBufferDirty = false;
+}
+
+template <typename TPixel, unsigned int VImageDimension>
+void
+CLEDataManager<TPixel, VImageDimension>::Update()
+{
+  this->UpdateGPUBuffer();
+  this->UpdateCPUBuffer();
+}
+
+template <typename TPixel, unsigned int VImageDimension>
+void
+CLEDataManager<TPixel, VImageDimension>::UpdateCPUBuffer()
+{
+  m_IsCPUBufferDirty = false;
+}
+
+template <typename TPixel, unsigned int VImageDimension>
+void
+CLEDataManager<TPixel, VImageDimension>::UpdateGPUBuffer()
+{
+  m_IsGPUBufferDirty = false;
+}
+
+template <typename TPixel, unsigned int VImageDimension>
+void
+CLEDataManager<TPixel, VImageDimension>::Graft(const CLEDataManager * data)
+{
+  //if (data)
+  //{
+  //  m_BufferSize = data->m_BufferSize;
+  //  m_ContextManager = data->m_ContextManager;
+  //  m_CommandQueueId = data->m_CommandQueueId;
+  //
+  //  if (m_GPUBuffer) // Decrease reference count to GPU memory
+  //  {
+  //    clReleaseMemObject(m_GPUBuffer);
+  //  }
+  //  if (data->m_GPUBuffer) // Increase reference count to GPU memory
+  //  {
+  //    clRetainMemObject(data->m_GPUBuffer);
+  //  }
+  //
+  //  m_GPUBuffer = data->m_GPUBuffer;
+  //
+  //  m_CPUBuffer = data->m_CPUBuffer;
+  //  //    m_Platform  = data->m_Platform;
+  //  //    m_Context   = data->m_Context;
+  //  m_IsCPUBufferDirty = data->m_IsCPUBufferDirty;
+  //  m_IsGPUBufferDirty = data->m_IsGPUBufferDirty;
+  //}
+}
+template <typename TPixel, unsigned int VImageDimension>
+void
+CLEDataManager<TPixel, VImageDimension>::PrintSelf(std::ostream & os, Indent indent) const
+{
+  //os << indent << "CLEDataManager (" << this << ")" << std::endl;
+  //os << indent << "m_BufferSize: " << m_BufferSize << std::endl;
+  //os << indent << "m_IsGPUBufferDirty: " << m_IsGPUBufferDirty << std::endl;
+  //os << indent << "m_GPUBuffer: " << m_GPUBuffer << std::endl;
+  //os << indent << "m_IsCPUBufferDirty: " << m_IsCPUBufferDirty << std::endl;
+  //os << indent << "m_CPUBuffer: " << m_CPUBuffer << std::endl;
```

Comment:
uncomment

We can exercise these and more with

```
ITK_EXERCISE_BASIC_OBJECT_METHODS
```

from *itkTestingMacros.h*.